core/src/main/java/com/graphhopper/routing/util/parsers/helpers/OSMValueExtractor.java

```diff
@@ -30,12 +30,8 @@ public static void extractTons(IntsRef edgeFlags, ReaderWay way, DecimalEncodedV
         final String rawValue = way.getFirstPriorityTag(keys);
         double value = stringToTons(rawValue);
 
-        if (Double.isNaN(value)) {
-            return;
-        }
+        if (Double.isNaN(value)) value = Double.POSITIVE_INFINITY;
 
-        if (value > valueEncoder.getMaxStorableDecimal())
-            value = valueEncoder.getMaxStorableDecimal();
```

Comment:
Ok, I see. We are trying to hide the fact that DecimalEncodedValue uses discrete values internally and sometimes even maps the highest value to infinity. Normally we round to the next discrete value and if `maximumIsInfinity==true` we cap finite values that are above the maximum to the highest finite value, which kind of makes sense. What is a bit ugly though is that this is not consistent to the behavior when `maximumIsInfinity==false`, because in this case we throw an error when the value is too large, which also makes sense, or even makes more sense. And wouldn't it be useful to warn or error in case the OSM value extractor returns a too large value? For example if we chose too few bits to store max_height, or if the values were too high because of mapping errors using wrong units etc. maybe we'd like to find out instead of silently truncating to the highest finite value? In this case we'd probably need something like `getMaxStorableFiniteDecimal()` or something as long as we keep hiding the internal representation. Or we could make `setDecimal` return the value that was actually stored, which would also allow printing such a warning (or throw an error even).

Reply:
>  Can you explain how you imagine the double setDecimal would solve this and how we differentiate between another intermediate value?

```java
double actualValue = decEnc.setDecimal(value);
if (Math.abs(actualValue - value) > threshold)
   throw new IllegalStateException("this did not work");
```